src/mono/mono/mini/mini.c

```diff
@@ -393,7 +393,11 @@ mono_type_to_load_membase (MonoCompile *cfg, MonoType *type)
 	case MONO_TYPE_I4:
 		return OP_LOADI4_MEMBASE;
 	case MONO_TYPE_U4:
+#ifdef TARGET_RISCV64
+		return OP_LOADI4_MEMBASE;
```

Comment:
cuz the psABI of RISCV hase specify that
> When passed in registers or on the stack, integer scalars narrower than XLEN bits are widened according to the sign of their type up to 32 bits, then sign-extended to XLEN bits.

which means the unsigned value on stack will be load by the same instruction as signed value. Following picture is generate by clang rv64 trunk https://godbolt.org/z/Tj3e3nYM6 as example.
<img width="491" alt="6f791ef3f9cb2643c1136d271e3f137" src="https://github.com/dotnet/runtime/assets/43398706/c693728f-9114-4d2c-806e-1b8a13fef383">

while `OP_LOADU4_MEMBASE` inst will bonly be used when zero extend.

Reply:
except that, the (uint32_t)-1 will be load into reg as `0xFFFFFFFFFFFFFFFF` in rv64 directly. if we use `lwu` to load -1 on stack, it will be  `0x00000000FFFFFFFF`. Thus, it will cause the -1 in reg doesn't equals to the -1 on stack.

Hope I have explain it clearly.

Reply:
does this conversation be solved? Since I explain why I write like that.